integration-tests/hibernate-validator-resteasy-reactive/src/main/java/io/quarkus/it/hibernate/validator/TestInterfaceRestClient.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.it.hibernate.validator;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.groups.ConvertGroup;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@RegisterRestClient
+@Path("/test")
+public interface TestInterfaceRestClient {
+    @POST
+    @Consumes({ "application/json" })
+    Response doSomething(@Valid @ConvertGroup(to = Error.class) @NotNull String myParameter);
```

Comment:
Should we have an actual test to make sure the constraint are validated? Not necessarily with this example.

Reply:
yeah, I'll try adding one, having this interface only tells us that the build will not fail if it is a part of the app 🙈 😃

Reply:
OK, I've pushed the test, but this actually won't work 😖 

if we normalize the `$$CDIWrapper` class and end up with an interface  -- the bean metadata collection is ignored on the HV side. (somewhat related to this PR https://github.com/hibernate/hibernate-validator/pull/1399) the class hierarchy for the interface is empty, and as a result nothing gets collected in this block
https://github.com/hibernate/hibernate-validator/blob/d2694eced589af7a279e366984e797fc026491e9/engine/src/main/java/org/hibernate/validator/internal/metadata/PredefinedScopeBeanMetaDataManager.java#L88-L103

At this point, it seems that it would be better to do this https://github.com/hibernate/hibernate-validator/pull/1428 on the HV side and keep things as they are in Quarkus or figure out how to break the cycle and skip copying the bean validation annotations to `$$CDIWrapper` classes.

Reply:
Yeah, I don't see an easy way to break the cycle without it being brittle.

So, yeah, let's fix it on the HV side, I suppose.

Reply:
ok 🤝 I'll keep this draft open then and once we upgrade HV with https://github.com/quarkusio/quarkus/pull/42292 I'll double check that the test in this PR passes...

Reply:
The upgrade https://github.com/quarkusio/quarkus/pull/42292 should fix it. Moved the test there. I am going to close this PR.